indexer/modules/custom/pendle/export_pendle_pools_job.py

```diff
@@ -0,0 +1,124 @@
+import logging
+from typing import List
+
+from web3 import Web3
+
+from indexer.domain.log import Log
+from indexer.domain.transaction import Transaction
+from indexer.executors.batch_work_executor import BatchWorkExecutor
+from indexer.jobs import FilterTransactionDataJob
+from indexer.modules.custom.pendle.abi.event import *
+from indexer.modules.custom.pendle.abi.function import *
+from indexer.modules.custom.pendle.domains.market import PendlePoolD
+from indexer.specification.specification import TopicSpecification, TransactionFilterByLogs
+from indexer.utils.multicall_hemera import Call
+from indexer.utils.multicall_hemera.multi_call_helper import MultiCallHelper
+
+logger = logging.getLogger(__name__)
+
+
+class PendlePoolsJob(FilterTransactionDataJob):
+    dependency_types = [Transaction]
+    output_types = [PendlePoolD]
+    able_to_reorg = True
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+        self._batch_work_executor = BatchWorkExecutor(
+            kwargs["batch_size"],
+            kwargs["max_workers"],
+            job_name=self.__class__.__name__,
+        )
+        self._is_batch = kwargs["batch_size"] > 1
+        self._filters = kwargs.get("filters", [])
+        self.multicall_helper = MultiCallHelper(
+            self._web3, {"batch_size": kwargs["batch_size"], "multicall": True, "max_workers": kwargs["max_workers"]}
+        )
+
+    def get_filter(self):
+        return [
+            TransactionFilterByLogs(
+                [
+                    TopicSpecification(
+                        addresses=[
+                            self.user_defined_config["pendle_market_factory_address"],
+                            self.user_defined_config["pendle_market_factory_v3_address"],
+                        ],
+                        topics=[create_market_event.get_signature()],
+                    )
+                ]
+            ),
+        ]
+
+    def _collect(self, **kwargs):
+        pass
+
+    def _process(self, **kwargs):
+        logs: List[Log] = self._data_buff.get(Log.type(), [])
+        if len(logs) == 0:
+            return
+        calls = []
+        pools = {}
+        for log in logs:
+            if log.address not in [
+                self.user_defined_config["pendle_market_factory_address"],
+                self.user_defined_config["pendle_market_factory_v3_address"],
+            ]:
+                continue
+            if log.topic0 not in [create_market_event.get_signature(), create_market_event_v3.get_signature()]:
+                continue
+            market_address = Web3.to_checksum_address("0x" + log.topic1[26:])
```

Comment:
We should have a util function for this

Reply:
t has already been implemented in the Lido pr. The code will be updated after the merge.